build-info-extractor-gradle/src/test/java/org/jfrog/gradle/plugin/artifactory/task/DeployTaskProxyTest.java

```diff
@@ -0,0 +1,242 @@
+
+        ProxyConfiguration result = artifactoryManager.getProxyConfiguration();
+        assertEquals(result.host, "proxy-host");
+        assertEquals(result.port, 9999);
+    }
+
+    @Test
+    public void givenExplicitProxyHostButNoPort_doesNotUseProxy() {
+        ArtifactoryClientConfiguration clientConfig = new ArtifactoryClientConfiguration(log);
+        clientConfig.proxy.setHost("proxy-host");
+
+        ArtifactoryManager artifactoryManager = new ArtifactoryManager("", log);
+
+        DeployTask.configureProxy("", clientConfig, artifactoryManager);
+
+        assertNull(artifactoryManager.getProxyConfiguration());
+    }
+
+    @Test
+    public void givenExplicitProxyPortButNoHost_doesNotUseProxy() {
+        ArtifactoryClientConfiguration clientConfig = new ArtifactoryClientConfiguration(log);
+        clientConfig.proxy.setPort(9999);
+
+        ArtifactoryManager artifactoryManager = new ArtifactoryManager("", log);
+
+        DeployTask.configureProxy("", clientConfig, artifactoryManager);
+
+        assertNull(artifactoryManager.getProxyConfiguration());
+    }
+
+    @Test
+    public void givenExplicitAndSystemProxyHostAndPort_usesExplicitProxyHostAndPort() {
+        System.setProperty(HTTPS_PROXY_HOST_SETTING, "wrong-proxy-host");
+        System.setProperty(HTTPS_PROXY_PORT_SETTING, String.valueOf(4444));
+
+        ArtifactoryClientConfiguration clientConfig = new ArtifactoryClientConfiguration(log);
+        clientConfig.proxy.setHost("proxy-host");
+        clientConfig.proxy.setPort(9999);
+
+        ArtifactoryManager artifactoryManager = new ArtifactoryManager("", log);
+
+        DeployTask.configureProxy("", clientConfig, artifactoryManager);
+
+        ProxyConfiguration result = artifactoryManager.getProxyConfiguration();
+        assertEquals(result.host, "proxy-host");
+        assertEquals(result.port, 9999);
+    }
+
+    @Test
+    public void givenSystemProxyHostAndPort_usesSystemProxyHostAndPort() {
+        System.setProperty(HTTPS_PROXY_HOST_SETTING, "proxy-host");
+        System.setProperty(HTTPS_PROXY_PORT_SETTING, String.valueOf(9999));
+
+        ArtifactoryClientConfiguration clientConfig = new ArtifactoryClientConfiguration(log);
+        ArtifactoryManager artifactoryManager = new ArtifactoryManager("", log);
+
+        DeployTask.configureProxy("", clientConfig, artifactoryManager);
+
+        ProxyConfiguration result = artifactoryManager.getProxyConfiguration();
+        assertEquals(result.host, "proxy-host");
+        assertEquals(result.port, 9999);
+    }
+
+    @Test
+    public void givenSystemProxyHostAndPort_whenContextUrlIsHttp_usesSystemHttpProxyHostAndPort() {
+        System.setProperty(HTTPS_PROXY_HOST_SETTING, "https-proxy-host");
+        System.setProperty(HTTPS_PROXY_PORT_SETTING, String.valueOf(1111));
+        System.setProperty(HTTP_PROXY_HOST_SETTING, "http-proxy-host");
+        System.setProperty(HTTP_PROXY_PORT_SETTING, String.valueOf(2222));
```

Comment:
The above line are duplicated many times in the code. I suggest we move them into a method and reuse it.